diff --git a/android/build.gradle b/android/build.gradle
index ed87f7388da6f82b99592a247c80befbb3f68acd..3e4c41d7898f118a1622fbf395a10ee3e5961be0 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -17,33 +17,9 @@ buildscript {
 
 apply plugin: 'com.android.library'
 
-def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION.tokenize('.')[0].toInteger()
-def shouldUseNameSpace = agpVersion >= 7
-def PACKAGE_PROP = "package=\"org.reactnative.maskedview\""
-def manifestOutFile = file("${projectDir}/src/main/AndroidManifest.xml")
-def manifestContent = manifestOutFile.getText()
-if(shouldUseNameSpace){
-      manifestContent = manifestContent.replaceAll(
-        PACKAGE_PROP,
-        ''
-    )  
-} else {
-    if(!manifestContent.contains("$PACKAGE_PROP")){
-        manifestContent = manifestContent.replace(
-            '<manifest',
-            "<manifest $PACKAGE_PROP "
-        )
-    }
-}
-manifestContent.replaceAll("  ", " ")
-manifestOutFile.write(manifestContent)
-
 android {
   compileSdkVersion safeExtGet('compileSdkVersion', 28)
-  
-  if(shouldUseNameSpace){
-    namespace = "org.reactnative.maskedview"
-  }
+  namespace = "org.reactnative.maskedview"
 
   defaultConfig {
     minSdkVersion safeExtGet('minSdkVersion', 16)
diff --git a/android/src/main/AndroidManifest.xml b/android/src/main/AndroidManifest.xml
index 5d1b95eed3202902e42f155179cca2cef0bb8117..a2f47b6057db70071e9080281db29d78ac165e66 100644
--- a/android/src/main/AndroidManifest.xml
+++ b/android/src/main/AndroidManifest.xml
@@ -1,2 +1,2 @@
-<manifest package="org.reactnative.maskedview" xmlns:android="http://schemas.android.com/apk/res/android">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
 </manifest>
