diff --git a/src/dom/dom-entry.tsx b/src/dom/dom-entry.tsx
index 43a920aa9d1ec3457b070aeaa9adcfbca844614d..48692e2b99abbbbef999ddb18a301b20ed9fa6a6 100644
--- a/src/dom/dom-entry.tsx
+++ b/src/dom/dom-entry.tsx
@@ -57,11 +57,6 @@ export function registerDOMComponent(AppModule: any) {
   function DOMComponentRoot(props) {
     // Props listeners
     const [marshalledProps, setProps] = React.useState(() => {
-      if (typeof window.$$EXPO_INITIAL_PROPS === 'undefined') {
-        throw new Error(
-          'Initial props are not defined. This is a bug in the DOM Component runtime.'
-        );
-      }
       return window.$$EXPO_INITIAL_PROPS;
     });
 
@@ -77,14 +72,18 @@ export function registerDOMComponent(AppModule: any) {
     }, [setProps]);
 
     const proxyActions = React.useMemo(() => {
-      if (!marshalledProps.names) return {};
+      if (!marshalledProps?.names) return {};
       // Create a named map { [name: string]: ProxyFunction }
       return Object.fromEntries(
-        marshalledProps.names.map((key) => {
+        marshalledProps?.names.map((key) => {
           return [key, ACTIONS[key]];
         })
       );
-    }, [marshalledProps.names]);
+    }, [marshalledProps?.names]);
+
+    if (!marshalledProps) {
+      return null;
+    }
 
     return <AppModule {...props} {...(marshalledProps.props || {})} {...proxyActions} />;
   }
diff --git a/src/dom/webview-wrapper.tsx b/src/dom/webview-wrapper.tsx
index 3ee2761fd9070c770ecb53ef4ade67ba7dac524d..568f06c0a4de52483117fb355334986a42ba83ca 100644
--- a/src/dom/webview-wrapper.tsx
+++ b/src/dom/webview-wrapper.tsx
@@ -117,15 +117,24 @@ const RawWebView = React.forwardRef<object, Props>(({ dom, filePath, ...marshalP
     ...dom,
     containerStyle: [containerStyle, debugZeroHeightStyle, dom?.containerStyle],
     onLayout: __DEV__ ? debugOnLayout : dom?.onLayout,
+    onLoad: () => {
+      emit({ type: '$$props', data: smartActions })
+    },
     injectedJavaScriptBeforeContentLoaded: [
       // On first mount, inject `$$EXPO_INITIAL_PROPS` with the initial props.
       `window.$$EXPO_INITIAL_PROPS = ${JSON.stringify(smartActions)};true;`,
-      dom?.matchContents ? getInjectBodySizeObserverScript() : null,
       dom?.injectedJavaScriptBeforeContentLoaded,
       'true;',
     ]
       .filter(Boolean)
       .join('\n'),
+    injectedJavaScript: [
+      dom?.matchContents ? getInjectBodySizeObserverScript() : null,
+      dom?.injectedJavaScript,
+      'true;',
+    ]
+      .filter(Boolean)
+      .join('\n'),
     ref: webviewRef,
     source,
     style: [
