diff --git a/dist/modal.js b/dist/modal.js
index 165cd55a960dfaccbb238807b17b0cc43e4f7bcb..b50d9fda50452b2f0d7ebd6388afc20d5486c632 100644
--- a/dist/modal.js
+++ b/dist/modal.js
@@ -403,15 +403,14 @@ export class ReactNativeModal extends React.Component {
         if (this.props.onSwipe) {
             console.warn('`<Modal onSwipe="..." />` is deprecated and will be removed starting from 13.0.0. Use `<Modal onSwipeComplete="..." />` instead.');
         }
-        DeviceEventEmitter.addListener('didUpdateDimensions', this.handleDimensionsUpdate);
-        if (this.state.isVisible) {
+        this.deviceEvent = DeviceEventEmitter.addListener('didUpdateDimensions', this.handleDimensionsUpdate);        if (this.state.isVisible) {
             this.open();
         }
         BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
     }
     componentWillUnmount() {
         BackHandler.removeEventListener('hardwareBackPress', this.onBackButtonPress);
-        DeviceEventEmitter.removeListener('didUpdateDimensions', this.handleDimensionsUpdate);
+        this.deviceEvent.remove();
     }
     componentDidUpdate(prevProps) {
         // If the animations have been changed then rebuild them to make sure we're
