diff --git a/node_modules/react-native-progress/Circle.js b/node_modules/react-native-progress/Circle.js
index 79f0a9e..3c1e3fa 100644
--- a/node_modules/react-native-progress/Circle.js
+++ b/node_modules/react-native-progress/Circle.js
@@ -169,6 +169,7 @@ export class ProgressCircle extends Component {
           {border ? (
             <Arc
               radius={size / 2}
+              fill={fill}
               startAngle={0}
               endAngle={(indeterminate ? endAngle * 2 : 2) * Math.PI}
               stroke={borderColor || color}
diff --git a/node_modules/react-native-progress/Shapes/Arc.js b/node_modules/react-native-progress/Shapes/Arc.js
index f913a56..0a25a13 100644
--- a/node_modules/react-native-progress/Shapes/Arc.js
+++ b/node_modules/react-native-progress/Shapes/Arc.js
@@ -87,6 +87,7 @@ export default class Arc extends Component {
       <Path
         d={path}
         strokeLinecap={strokeCap}
+        fill="none"
         strokeWidth={strokeWidth}
         {...restProps}
       />